IPython/core/oinspect.py

```diff
@@ -936,6 +937,27 @@ def _info(self, obj, oname='', info=None, detail_level=0) -> dict:
 
         return object_info(**out)
 
+    @staticmethod
+    def _source_contains_docstring(src, doc):
+        """
+        Check whether the source *src* contains the docstring *doc*.
+
+        This is is helper function to skip displaying the docstring if the
+        source already contains it, avoiding repetition of information.
+        """
+        try:
+            def_node, = ast.parse(dedent(src)).body
+            node_docstring = ast.get_docstring(def_node)
+        except Exception:
+            # The source can become invalid (because it is re-fetched
+            # from the source file) so the parse can fail in arbitrary
```

Comment:
Right now you're catching `TypeError`s from `src` being `None` here, which is nothing to do with `parse` failing (see above comment)